Tele op update

tubaDude99 opened the pull request:
Before issuing a pull request, please see the contributing page.

tubaDude99 committed:
Autonomous cleanup

Deleting obsolete yaml files and Auto classes (I didn't delete the Auto Camera classes because I wasn't sure if we'd want to reuse the code)

tubaDude99 committed:
Create TeleOp_Basic.java

Created a basic TeleOp mode that just has simple movement (with precision mode for testing)

Aman-Anas reviewed (changes requested):
- Deadzone code is necessary to prevent controller imperfections from causing the robot to move slightly when idle, please keep (though you can set the dead zone range to something very small)
- z variable is for Z-axis rotation (robot orientation), X variable is for strafing axis. Apply power multiplier to all axes (line 69)
- Z-axis is still slowed down by almost half due to (z/1.8) section on line 106

tubaDude99 committed:
Update TeleOp_Basic.java

Added multiplier for turn speed (independent of linear speed), added stick deadzones

Aman-Anas commented:
OK so for deadzone code we need to remap the ranges when using a deadzone

So for example if the deadzone is 5 and the max is 100, then with the current code you can only move from 5-100 since as soon as you exit the deadzone you'll be at 5.1

Old deadzone code fixed that by mapping 0-100 onto the 5-100 range

Aman-Anas reviewed (changes requested):
Remap deadzone range

tubaDude99 committed:
Update TeleOp_Basic.java

Added code to adjust inputs to correct for the dead zone.

tubaDude99 committed:
More auto cleanup

Aman-Anas reviewed (approved):
Looks good, let's test this code next meeting (Thursday) before merging.

team3916 committed:
Fix Gradle configuration and build issues

team3916 committed:
Refactor code, remove old classes

team3916 committed:
Clean up old classes, restructure vision code
